SDVModTest/Options/ModOptionsPageButton.cs

```diff
@@ -31,39 +27,37 @@ public ModOptionsPageButton()
             xPositionOnScreen = activeClickableMenu.xPositionOnScreen + activeClickableMenu.width - 200;
             yPositionOnScreen = activeClickableMenu.yPositionOnScreen + 16;
             Bounds = new Rectangle(xPositionOnScreen, yPositionOnScreen, width, height);
-            ControlEvents.MouseChanged += OnMouseChanged;
-            ControlEvents.ControllerButtonPressed += ControlEvents_ControllerButtonPressed;
-            MenuEvents.MenuChanged += MenuEvents_MenuChanged;
+            events.Input.ButtonPressed += OnButtonPressed;
+            events.Display.MenuChanged += OnMenuChanged;
         }
 
-        private void MenuEvents_MenuChanged(object sender, EventArgsClickableMenuChanged e)
+        /// <summary>Raised after a game menu is opened, closed, or replaced.</summary>
+        /// <param name="sender">The event sender.</param>
+        /// <param name="e">The event arguments.</param>
+        private void OnMenuChanged(object sender, MenuChangedEventArgs e)
         {
-            if (Game1.activeClickableMenu is GameMenu)
+            if (e.NewMenu is GameMenu menu)
             {
-                GameMenu menu = Game1.activeClickableMenu as GameMenu;
                 xPositionOnScreen = menu.xPositionOnScreen + menu.width - 200;
             }
         }
 
-        private void ControlEvents_ControllerButtonPressed(object sender, EventArgsControllerButtonPressed e)
+        /// <summary>Raised after the player presses a button on the keyboard, controller, or mouse.</summary>
+        /// <param name="sender">The event sender.</param>
+        /// <param name="e">The event arguments.</param>
+        public void OnButtonPressed(object sender, ButtonPressedEventArgs e)
         {
-            if (e.ButtonPressed == Buttons.A &&
-                isWithinBounds(Game1.getMouseX(), Game1.getMouseY()))
+            if (e.Button == SButton.MouseLeft || e.Button == SButton.ControllerA)

```

Comment:
This is awesome! It's so nice that I can just ask if the mouse was pressed now. Thank you!

Reply:
Glad you like it!